src/main/scala/it/pagopa/interop/partyregistryproxy/service/impl/IPAOpenDataServiceImpl.scala

```diff
@@ -69,9 +82,19 @@ object IPAOpenDataServiceImpl {
     response.records.flatMap { record =>
       for {
         id             <- record(mapped(InstitutionsFields.taxCode)).select[String]
-        originId       <- record(mapped(InstitutionsFields.originId)).select[String]
+        originId       <- institutionKind match {
+          case InstitutionKind.Agency => record(mapped(InstitutionsFields.originId)).select[String]
+          case InstitutionKind.AOO    => record(mapped(InstitutionsFields.aooId)).select[String]
+          case InstitutionKind.UO     => record(mapped(InstitutionsFields.uoId)).select[String]
+        }
         taxCode        <- mapped.get(InstitutionsFields.taxCode).flatMap(idx => record(idx).select[String])
-        category       <- mapped.get(InstitutionsFields.category).flatMap(idx => record(idx).select[String])
+        category       <- institutionKind match {
+          case InstitutionKind.Agency =>
+            mapped.get(InstitutionsFields.category).flatMap(idx => record(idx).select[String])
+          case _                      =>
+            val institutionOriginId = record(mapped(InstitutionsFields.originId)).select[String]
```

Comment:
Despite it should not happen, this is safer
```suggestion
            val institutionOriginId = mapped.get(InstitutionsFields.originId).flatMap(idx => record(idx).select[String])
```